### ISSUE TITLE
Sonoff RF Bridge integration

### ISSUE THREAD

**Issue by peterbuga** [peterbuga/HASS-sonoff-ewelink#38]
This dumps contain one single _[Wireless Door Window Detector](https://www.banggood.com/DY-MC100B-Wireless-Door-Window-Detector-Sensor-Alarm-433MHz-for-Security-Alarm-System-p-1266532.html?rmmds=myorder&cur_warehouse=CN)_ 

Open door detector
```
[ { "__v": 0, "_id": "[hidden]" "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "brandName": "Sonoff", "createdAt": "xxxx-xx-xxxxx:xx:xx.xxx", "deviceStatus": "", "deviceUrl": "", "deviceid": "[hidden]" "devicekey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "extra": { "_id": "[hidden]" "extra": { "apmac": "xx:xx:xx:xx:xx:xx", "brandId": "58e5f344baeb368720e25469", "description": "WO1852", "mac": "xx:xx:xx:xx:xx:xx", "manufacturer": "\u6df1\u5733\u677e\u8bfa\u6280\u672f\u6709\u9650\u516c\u53f8", "model": "PSC-B67-GL", "modelInfo": "5a2e1ae50cf772f92c342ef6", "ui": "\u529f\u7387\u68c0\u6d4b\u63d2\u5ea7\u8fc7\u8f7d\u544a\u8b66", "uiid": 32 } }, "group": "", "groups": [], "ip": "[hidden]" "location": "", "name": "[hidden]", "offlineTime": "xxxx-xx-xxxxx:xx:xx.xxx", "online": true, "onlineTime": "xxxx-xx-xxxxx:xx:xx.xxx", "params": { "alarmCValue": [ -1, -1 ], "alarmPValue": [ -1, -1 ], "alarmType": "pcv", "alarmVValue": [ -1, -1 ], "controlType": 8, "current": "0.16", "endTime": "", "fwVersion": "2.8.0", "hundredDaysKwh": "get", "init": 1, "oneKwh": "get", "partnerApikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "power": "21.90", "rssi": -60, "sledOnline": "on", "staMac": "xx:xx:xx:xx:xx:xx", "startTime": "xxxx-xx-xxxxx:xx:xx.xxx", "switch": "on", "timeZone": 2, "voltage": "235.62" }, "productModel": "Pow_R2", "settings": { "alarmNotify": 1, "opsHistory": 1, "opsNotify": 0 }, "sharedTo": [], "showBrand": true, "type": "10", "uiid": 32 }, { "__v": 0, "_id": "[hidden]" "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "brandName": "Sonoff", "createdAt": "xxxx-xx-xxxxx:xx:xx.xxx", "deviceStatus": "", "deviceUrl": "", "deviceid": "[hidden]" "devicekey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "extra": { "_id": "[hidden]" "extra": { "apmac": "xx:xx:xx:xx:xx:xx", "brandId": "58e5f344baeb368720e25469", "description": "WO1851", "mac": "xx:xx:xx:xx:xx:xx", "manufacturer": "\u6df1\u5733\u677e\u8bfa\u6280\u672f\u6709\u9650\u516c\u53f8", "model": "PSF-BRA-GL", "modelInfo": "5a2e1b360cf772f92c342f01", "ui": "RFBridge", "uiid": 28 } }, "group": "", "groups": [], "ip": "[hidden]" "location": "", "name": "[hidden]", "offlineTime": "xxxx-xx-xxxxx:xx:xx.xxx", "online": true, "onlineTime": "xxxx-xx-xxxxx:xx:xx.xxx", "params": { "cmd": "trigger", "fwVersion": "2.7.0", "init": 1, "rfChl": 0, "rfList": [ { "rfChl": 0, "rfVal": "331801CC0582EE4D13" } ], "rfTrig0": "xxxx-xx-xxxxx:xx:xx.xxx", "rssi": -37, "sledOnline": "on", "timers": [] }, "productModel": "RF_Bridge", "settings": { "alarmNotify": 1, "opsHistory": 1, "opsNotify": 0 }, "sharedTo": [], "showBrand": true, "tags": { "disable_timers": [], "zyx_info": [ { "buttonName": [ { "0": "Button Name1" } ], "name": "[hidden]", "remote_type": "6" } ] }, "type": "10", "uiid": 28 } ] 
```

Closed door detector
```
[ { "__v": 0, "_id": "[hidden]" "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "brandName": "Sonoff", "createdAt": "xxxx-xx-xxxxx:xx:xx.xxx", "deviceStatus": "", "deviceUrl": "", "deviceid": "[hidden]" "devicekey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "extra": { "_id": "[hidden]" "extra": { "apmac": "xx:xx:xx:xx:xx:xx", "brandId": "58e5f344baeb368720e25469", "description": "WO1852", "mac": "xx:xx:xx:xx:xx:xx", "manufacturer": "\u6df1\u5733\u677e\u8bfa\u6280\u672f\u6709\u9650\u516c\u53f8", "model": "PSC-B67-GL", "modelInfo": "5a2e1ae50cf772f92c342ef6", "ui": "\u529f\u7387\u68c0\u6d4b\u63d2\u5ea7\u8fc7\u8f7d\u544a\u8b66", "uiid": 32 } }, "group": "", "groups": [], "ip": "[hidden]" "location": "", "name": "[hidden]", "offlineTime": "xxxx-xx-xxxxx:xx:xx.xxx", "online": true, "onlineTime": "xxxx-xx-xxxxx:xx:xx.xxx", "params": { "alarmCValue": [ -1, -1 ], "alarmPValue": [ -1, -1 ], "alarmType": "pcv", "alarmVValue": [ -1, -1 ], "controlType": 8, "current": "0.16", "endTime": "", "fwVersion": "2.8.0", "hundredDaysKwh": "get", "init": 1, "oneKwh": "get", "partnerApikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "power": "21.87", "rssi": -60, "sledOnline": "on", "staMac": "xx:xx:xx:xx:xx:xx", "startTime": "xxxx-xx-xxxxx:xx:xx.xxx", "switch": "on", "timeZone": 2, "voltage": "232.78" }, "productModel": "Pow_R2", "settings": { "alarmNotify": 1, "opsHistory": 1, "opsNotify": 0 }, "sharedTo": [], "showBrand": true, "type": "10", "uiid": 32 }, { "__v": 0, "_id": "[hidden]" "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "brandName": "Sonoff", "createdAt": "xxxx-xx-xxxxx:xx:xx.xxx", "deviceStatus": "", "deviceUrl": "", "deviceid": "[hidden]" "devicekey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "extra": { "_id": "[hidden]" "extra": { "apmac": "xx:xx:xx:xx:xx:xx", "brandId": "58e5f344baeb368720e25469", "description": "WO1851", "mac": "xx:xx:xx:xx:xx:xx", "manufacturer": "\u6df1\u5733\u677e\u8bfa\u6280\u672f\u6709\u9650\u516c\u53f8", "model": "PSF-BRA-GL", "modelInfo": "5a2e1b360cf772f92c342f01", "ui": "RFBridge", "uiid": 28 } }, "group": "", "groups": [], "ip": "[hidden]" "location": "", "name": "[hidden]", "offlineTime": "xxxx-xx-xxxxx:xx:xx.xxx", "online": true, "onlineTime": "xxxx-xx-xxxxx:xx:xx.xxx", "params": { "cmd": "trigger", "fwVersion": "2.7.0", "init": 1, "rfChl": 0, "rfList": [ { "rfChl": 0, "rfVal": "331801CC0582EE4D13" } ], "rfTrig0": "xxxx-xx-xxxxx:xx:xx.xxx", "rssi": -37, "sledOnline": "on", "timers": [] }, "productModel": "RF_Bridge", "settings": { "alarmNotify": 1, "opsHistory": 1, "opsNotify": 0 }, "sharedTo": [], "showBrand": true, "tags": { "disable_timers": [], "zyx_info": [ { "buttonName": [ { "0": "Button Name1" } ], "name": "[hidden]", "remote_type": "6" } ] }, "type": "10", "uiid": 28 } ] 
```

**Comment by DenielTorres** [comment]
Hi @peterbuga! Thanks for opening this "issue" as well. Let me know if I can help with this integration! I'm actually about to order a bunch of more from these sensors for all windows and doors at my home. 
Good luck!

**Comment by peterbuga** [comment]
@DenielTorres i've been watching this video https://www.youtube.com/watch?v=JP1X4wOjlTk and it turn out it might be a bit more tricky view the state of RF devices attached to the Sonoff RF Bridge, mostly because and RF device might change its ON/OFF state very fast and might not get picked up by the way the component works inside HA.
Luckily I already have some ideas on how to achieve this, but this also sort-of implies that I might bug you for a while to do all kinds of tests. If you're up for it I'll do my best and start working on it 😉 

ps: I don't know if you knew this but turns out a Sonoff RF Bridge supports only 4 RF devices max! it's mentioned in the video and also here http://ewelink.coolkit.cc/?p=886 that's kinda of a small let down 😞 I thought I should let you know before you "_order a bunch of more from these sensors for all windows and doors at my home_"

**Comment by max-rh** [comment]
Hi @peterbuga , so sorry for the delay....and thanks for the updated debug script :).
Here is the dumps, one for the RF_Bridge, and for the Bulb, and the third one is for my 2 Gang Generic Wall Switch, i know you did add support for 1 and 3 gangs, but this 2-gang shows up as 4 buttons on my HA, i think that the 3rd button is for an all channels on/off but i don't know whats the 4th one for,,,,

Happy to be of assists and thank a lot for your support 
[Bulb.txt](https://github.com/peterbuga/HASS-sonoff-ewelink/files/2751315/Bulb.txt)
[2-Button.txt](https://github.com/peterbuga/HASS-sonoff-ewelink/files/2751316/2-Button.txt)
[RF_Bridge.txt](https://github.com/peterbuga/HASS-sonoff-ewelink/files/2751317/RF_Bridge.txt)

**Comment by peterbuga** [comment]
@max-rh thanks for the requested data!
looks like you have only 1 RF device attached to the Bridge, please let me know what kind it is: door / motion detector etc, i'll need to create some specific tests for RF devices because I believe they might react very fast between ON/OFF states, currently incompatible with `master` branch 

regarding the *2-gang showing as 4*, in eWeLink app there's 2 types of devices: single-switch devices and multi-switch devices. *all* the multi switch-devices have 4 "buttons", it doesn't matter if physically they are 2 or 3 :| the eWeLink app has thousands of devices lists to figure out what device has what, it's a real horror show wrapped in a spaghetti code which i'm not going to replicate!
the `websocket` branch (which is awesome btw compared to the current `master`) tries to _guess_ the number of switches and works ok in most of the cases, but it's currently not on par with latest `master` additions. i haven't made it the default one yet, because i expect a possible wave of complaints that i don't want to take care right now 😅 also this branch is going to be the answer for the RF devices (hopefully 🤞)

**Comment by max-rh** [comment]
Great, one issue at a time. Thanks peter.
Regarding the RF sensor, its a Door sensor, I do have couple of motion sensors to be integrated later, will update u with their dumps as soon as i do them. and i plan to get few other rf devices.

Regarding the ON/OFF state, i am not sure i get it. but i dont think the sensors react/send any data when u close it, it only sends an alarm when it opens.

**Comment by peterbuga** [comment]
> I do have couple of motion sensors to be integrated later, will update u with their dumps as soon as i do them

i'm interested in a dump that has more RF devices attached to the sonoff bridge, to see exactly how they are laid out (i know my requests might sound a bit weird, but eWeLink coders or itead company or whomever makes this devices for some reason choose to add different logic for different devices and in some cases for different firmwares too. i tried to guess them them before with little data that i had but it has just a matter of time till someone threw me a bug 😓)
please add a key fob (if you have one) i'm curious how this one is represented and how to ignore it, or maybe i'll add it too i don't know, i don't see a big usage to manually trigger HA actions or similar...to be seen.

> Regarding the ON/OFF state, i am not sure i get it. but i dont think the sensors react/send any data when u close it, it only sends an alarm when it opens.

right now, with `master` branch, you get updates every 10-15sec or whatever you set up as `scan_interval`. a so called "alarm" is activated when (door gets open) a RF device goes from 0 to 1 (or maybe viceversa 😅) and silently changes its state back (when you close the door) or so i think, we'll figure it out in the end. if this changes happen between the defined `scan_interval` period they will not be picked up acting like nothing happened.

**Comment by max-rh** [comment]
Got it, Yeah i think your right, but its worth testing.
Regarding the sensors, i installed the motion sensor, but sadly i don't have a key fob, here is what the dump looks like with the two sensors. (I am expecting couple more rf devices in the next few week).
[RF_Bridge.txt](https://github.com/peterbuga/HASS-sonoff-ewelink/files/2752026/RF_Bridge.txt)

**Comment by peterbuga** [comment]
I think I have what I need so far 👍 I just need to find the time to crunch some code out of all this info.
I'll let you know when I'll need some (inevitable) testing.

Thanks again for support!

**Comment by aussiebyrd** [comment]
Hi @peterbuga - in case another dump help - here is mine. My Sonoff RF bridge is connected to two door sensors (Sonoff DW1), a DC fan (configured with three fan speeds and a fan-off option) and a DC fan with LED light (configured with light on/off, two fan speeds and a fan-off option).
A mains-power Sonoff switch (S26 - AU version, already working in HA thanks to you) is also in the dump.

Let me know if I can help at all.
[SonoffRF_Bridge.txt](https://github.com/peterbuga/HASS-sonoff-ewelink/files/2882595/SonoffRF_Bridge.txt)

**Comment by nzdazza** [comment]
Hello @peterbuga, I am very keen to get the alarm triggers for the RF Bridge working too. From my perspective:
Do care about receiving alarm triggers
4 triggers per device limit - awful, guess I'll just be buying a few...  
Missing triggers < 10-15 sec apart, not worried. As long as the initial trigger is received I don't care about the "off".
I am using cheap wireless PIR detectors which are picked up by the sonoff just fine

Happy to help in any way.

The out-of-the-box approach, despite the reliance on the cloud, is working very well for me so far. Love your work!

**Comment by peterbuga** [comment]
@nzdazza it'll be a cool nice/help if you generate a new debug log using the new extended method as defined in the current `master` branch. all you need to do is to read carefully the instructions on how to activate and generate the `debug` output (it's all mentioned in the project's description)

basically, after setting it up i need you to toggle on/off or change whatever settings you can related to Sonoff RF Bridge so that I can see what kind of messages are received over websocket. the previous approach gave me a "static" version of what is happening and I could only guess what I might be able to receive. 
ps: no worries if it's too hard or can't do it for whatever reason, i'll figure it out at some point 😉

**Comment by martusi61** [comment]
Dear Peter, in advance thank you for the work you are doing, then I wanted to know where you are with the integration of Sonoff RF, I have already bought 2 with some sensors and soon I will have to increase them. If you need help, debugging or anything else, I make myself available. Thanks again and good work.

**Comment by FixxCZ** [comment]
Just BTW for those who are not aware, there is a way how to use Sonoff bridge to translate any RF signal to MQTT. https://github.com/1technophile/OpenMQTTGateway/wiki/Sonoff-RFB 
I'm using it the way that door sensor and motion sensor via this gateway controling the light with Sonoff RF via MQTT and this custom component. And I have RF wall switch that is paired with Sonoff RF directly (using stock firmware). That means that automation controls turning off and on based on movement, timeout. But wall switch still ultimately control the relay and can override any other control.

**Comment by martusi61** [comment]
Thanks FixxCZ, I already knew it, but I was interested in using Sonoff WITHOUT changing the original firmware, so I really appreciate Peter's work, thanks to him I already use about 20 Sonoff switches on HomeAssistant.

**Comment by FreakyKiwi** [comment]
Hi Peter, 
Thanks for the great work - I have successfully added several Sonoff switches to my new Home Assistant installation.
I am using the Sonoff RF Bridge 433 in the opposite way to the issue here and am looking for support in HA.
I am triggering the bridge via WiFi to send RF commands out to control my motorized projector screen.

The Bridge does not currently appear in HA at all although it is linked to the same account as my other power switches which do appear and work perfectly.

The Bridge supports four captured RF codes of which I am using three as per the log below (Down/Stop/Up)

Hopefully this can be supported and added to HA which will allow me to expose the switches via Hue emulation and include in my Harmony Hub config.

2019-06-22 06:52:53.901 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "1000665e0f", "params": {"cmd": "transmit", "rfChl": 0}, "sequence": "1561186372131", "tempRec": "1000665e0f", "ts": 0, "userAgent": "app"}

2019-06-22 06:52:56.248 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "1000665e0f", "params": {"cmd": "transmit", "rfChl": 1}, "sequence": "1561186374506", "tempRec": "1000665e0f", "ts": 0, "userAgent": "app"}

2019-06-22 06:52:58.261 [W] {"action": "update", "apikey": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", "deviceid": "1000665e0f", "params": {"cmd": "transmit", "rfChl": 2}, "sequence": "1561186376545", "tempRec": "1000665e0f", "ts": 0, "userAgent": "app"}

**Comment by faisal-qureshi** [comment]
> The Bridge does not currently appear in HA at all although it is linked to the same account as my other power switches which do appear and work perfectly.

I have the same problem.  I don't see my RF bridge, or the Sonoff Bulbs, in Home Assistant. Do I need to do something special for them to appear as sensors/switches, or something?

**Comment by peterbuga** [comment]
> I have the same problem

it's not a problem, it's just not implemented. (otherwise this issue would not be open)

> or the Sonoff Bulbs

there's another branch with support for sonoff bulb https://github.com/peterbuga/HASS-sonoff-ewelink/tree/sonoff-bulb also please take a look a the (pretty long) implementation of it and see what works and doesn't https://github.com/peterbuga/HASS-sonoff-ewelink/issues/34 before asking/requesting further infos

**Comment by FreakyKiwi** [comment]
Thanks Peter,
Do you have any plans to add support this device or should I look at flashing with Tasmota or using another supported RF transmitter?

Let me know if there’s anything I can do to help.

Matt

**Comment by peterbuga** [comment]
@FreakyKiwi the plan was to add as many devices as possible but it's unbelievable hard to pseudo-blindly code when i don't have the device to test it and everything requires a tons of back & forth messages to see what i'm doing wrong, translated in time that I don't quite have it right now/anymore (also, it's summer right now and i just hate to stay inside and code 😅).
in the case of sonoff bridge i think tasmota it's a better fit because you can attach many more rf devices to it vs. the original fw (unless itead changed something meanwhile, i'm not totally up to date) so it's entirely up to you. 

i'll eventually add it, i just can't say when `¯\_(ツ)_/¯`

**Comment by martusi61** [comment]
Peter, thank you for everything you've done so far, I'll follow your advice, for the Sonoff Bridge RF I'll use the Tasmota firmware.

**Comment by ranski1999** [comment]
I have ewelink integrated on home assistant with original firmware and I have a RF Bridge and I cannot see it in home assistant with the door sensors I have
 Can someone help please

**Comment by peterbuga** [comment]
> Can someone help please

@ranski1999 can you please read the thread?

**Comment by ranski1999** [comment]
Thanks???

**Comment by byxil** [comment]
hello peterbuga, thanks for the work you've done.
I'm also interested in using sensors without using the tasmota firmware.
if you need help to do some tests, I'm available (byxil1 @ gmail.com), I have a presence sensor and door contact.

**Comment by peterbuga** [comment]
good news 🙌 some generous people sent a small donation and i bought a RF Bridge + a sensor, not sure when they'll be shipped exactly (probably in < 1month `¯\_(ツ)_/¯`) but this will be a next device to be added 👍

**Comment by LiuVik** [comment]
Hi Peter, thank you for yours good work.
I have a question for you.... i used installation of your script in accordance with your manual, step by step. But, in HA i seen like results only relay swithches... my RF Bridge do not want to appear in dashboard or in devices in to the HA...
Help me please...
All actions in accordance with manual

**Comment by peterbuga** [comment]
@LiuVik this is going to sound mean but there’s a reason this issue still exists + only if you took 10sec to read the previous comment you’ll have your answer ... 😑

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by peterbuga** [comment]
hi anyone/everyone 
i just got my hands on a RF Bridge / remote / motion detector 🙌 i'm looking forward to integrate this into HA in the feature, but i have a small logical problem: i don't know exactly how i should represent RF bridge connected devices into HA :(

for example, every time motion is detected just a short-timed feedback is sent via itead servers to the ewelink app which in turn triggers the _alarm notification_ in the mobile app (and the scenes it contains as a trigger). i'm not familiar with any other alarm-like system in HA and/or how they behave and any input would be appreciated.

so far, the best i could come up is to represent each alarm-like device/ remote-buttons as a `binary_sensor` that would stay ON for like 1-5 seconds enough for the HA automations to capture and execute the rules based on it.

**Comment by faisal-qureshi** [comment]
Why even 5seconds?

An instantaneous high would trigger any script/automation

**Comment by peterbuga** [comment]
@faisal-qureshi i never played with 'instant high' in HA automations so i'm not sure how well they'll behave. as far as i know HA checks every 1sec the entities status so I flip the binary_sensor ON/OFF in less than 1sec i'm afraid it'll be missed

**Comment by faisal-qureshi** [comment]
Oh shoot. You're right.  I forgot that this is pulling statuses. In which
case it could miss even on the 5 second.

This IS tricky.

**Comment by peterbuga** [comment]
i've merge a bunch of new things in a new dev branch https://github.com/peterbuga/HASS-sonoff-ewelink/tree/dev and there's support for RF Bridge too

remote buttons act and they are displayed almost in same way code-wise, the only difference is that alarms are triggering well... alarm notifications.
new `binary_sensor` entities will be created for each button/alarm connected to the RF Bridge and they'll automatically turn off after ~2sec, enough to trigger an automation

it's still in development so keep an eye on the log errors!

**Comment by ranski1999** [comment]
Thanks I have been waiting for this great job.

**Comment by peterbuga** [comment]
@ranski1999 did you tried it yet?

**Comment by ranski1999** [comment]
Yes it work well for sensors have not figured out plugs. I have 433 kHz plugs, they show up as binary sensors. On my phone if I open ewelink and home assistant together all of my sonoff devices go to unavailable about one minute after but if I open one at a time it works just fine. But overall it works great.

**Comment by peterbuga** [comment]
> On my phone if I open ewelink and home assistant together all of my sonoff devices go to unavailable about one minute

you need to use the 2nd ewelink account & share devices to avoid this problem (if you already do this I do not know what is the problem right now 😓)

>  I have 433 kHz plugs, they show up as binary sensors.

when/if you have time could you please enable the `debug` mode of this component, follow the instructions and post the output. i personally have only RF Bridge + crappy remote control + motion detector. i don't have 433khz slampher/plugs/switch that can accept an RF signal and I don't know how they work exactly :| or for that matter how they are displayed in eWeLink app (alternatively if possible show me a phone screenshot of the bridge devices, might shed some light for me)

**Comment by ranski1999** [comment]
![Screenshot_20191104-045801](https://user-images.githubusercontent.com/51892016/68116117-0c9c6080-fec8-11e9-9cd3-7347033d3f2e.png)

**Comment by peterbuga** [comment]
@ranski1999 i have to say i'm a bit confused about your setup exactly 😬 
are the `Plug N` the physical plug switches or they are the buttons of a RF control remote? can you please link what exactly are the type/model of your plugs?

but more than that i think i need the dump of the `debug` to get the whole picture ...

**Comment by martusi61** [comment]
Very good Peter, thank you for your great commitment to all of us, I then followed your advice and uploaded Tasmota to my bridge and discovered another planet! To be free, To be free, To be free!

**Comment by ranski1999** [comment]
![20191104_083525](https://user-images.githubusercontent.com/51892016/68124959-863f4900-fede-11e9-9b4b-8df859dc584a.jpg)

![20191104_083659](https://user-images.githubusercontent.com/51892016/68124977-90f9de00-fede-11e9-90c2-7860e94cb33d.jpg)

**Comment by ranski1999** [comment]
![developer](https://user-images.githubusercontent.com/51892016/68125340-60667400-fedf-11e9-872d-ad81d230a5e6.png)

**Comment by ranski1999** [comment]
Sorry with all the photos. 
Now how do I do debug mode.
Is it in home asstants or in ewelink app.

**Comment by ranski1999** [comment]
I made the 2nd ewelink account and shared like you said and I worked really good so far has been working for 1 hour.

**Comment by peterbuga** [comment]
@ranski1999 good and less-good news, i sort of figured out what kind of plug devices you have (the screenshot were good 👍)

one last question before i draw my last conclusion, can you please let me know that you can toggle ON/OFF a plug from the eWeLink app? i don't have a device to receive a RF signal nor a feedback from app so i cannot tell it for sure :( 
basically i need to know if RF Bridge can also send RF signals out not just receive

**Comment by ranski1999** [comment]
Yes you press the button on the left it turns on the plug. The button on the right turns it off. It is setup as a button device

**Comment by ranski1999** [comment]
![Screenshot_20191104-164738](https://user-images.githubusercontent.com/51892016/68161172-4484c180-ff23-11e9-832a-f38a45612634.png)

**Comment by ranski1999** [comment]
Here is a video that he sets it up. Go to 8:20 that were the 433 plug is. I hope this all helps you. I appreciate all of this that you are doing to make this work.👍

https://youtu.be/JP1X4wOjlTk

**Comment by peterbuga** [comment]
@ranski1999 ok ok i think i understand it now 😅 i know some of my questions might sound a bit stupid but i'm kinda stuck in a corner sometimes trying to understand how they work - the RF Bridge basically replicates the remote rf buttons.

- in a remote case i will create momentary buttons. not sure if i can define them to be under the same  HA card you might have to find a way to _merge_ them if this is a proper way to explain it
- when the rf is defined as an alarm inside eWeLink they'll be shown just like now as binary_sensors
- i'll probably add the curtain option too but i'm not even sure if these devices even exist

**Comment by ranski1999** [comment]
That sounds really good if you need me to test something for you I will I'm just glad I'm able to help. I'm not complaining because this is awesome.
I was wondering if there was a way that it could be changed.
I used it for a couple days I think in home assistant it should say open or close not on or off.
And remember I'm not complaining I could live with it

**Comment by peterbuga** [comment]
@ranski1999 these are the HA guidelines https://www.home-assistant.io/integrations/binary_sensor/ specifically:

> door: On means open, Off means closed
> window: On means open, Off means closed

you could create/duplicate the binary_switches to have a sensor with open/close values with the template platform. i won't go into details about this, it's all in the documentation ;)

**Comment by ranski1999** [comment]
Thank you I found that I can use the customization in home assistant. And I use the change node in node red for my notifications and now they all say open for on and close for off.

**Comment by peterbuga** [comment]
@ranski1999 & everyone else
the update on RF Bridge to send-out-signals from HA aka the "rf buttons" (not just receive as it's currently now) it'll be delayed a bit due to the fact that the more devices I add the messier the code becomes and I need to re-write everything to accommodate this volatile type of API changes.

in the case of triggering to send rf signals from HA, the request itself is totally different than what it's used until now and i just cannot continue with the current code :pensive:

**Comment by CptKalibak** [comment]
> @ranski1999 & everyone else
> the update on RF Bridge to send-out-signals from HA aka the "rf buttons" (not just receive as it's currently now) it'll be delayed a bit due to the fact that the more devices I add the messier the code becomes and I need to re-write everything to accommodate this volatile type of API changes.
> 
> in the case of triggering to send rf signals from HA, the request itself is totally different than what it's used until now and i just cannot continue with the current code

Hi, 
Thanks for your great work and enthusiasm.
I'm looking forward to seeing the RF bridge work in HA. It would be a game changer for me. 

Just wanted the tell my gratitude and appreciation for your hard work on this!

**Comment by LukaszP2** [comment]
> in the case of sonoff bridge i think tasmota it's a better fit because you can attach many more rf devices to it vs. the original fw (unless itead changed something meanwhile, i'm not totally up to date) so it's entirely up to you.
> 
Now it can be 16 x 4 device in Sonoff rf

**Comment by ashoktvm** [comment]
Is RF bridge updated to 3.3.0 supported?

**Comment by ashoktvm** [comment]
> > in the case of sonoff bridge i think tasmota it's a better fit because you can attach many more rf devices to it vs. the original fw (unless itead changed something meanwhile, i'm not totally up to date) so it's entirely up to you.
> 
> Now it can be 16 x 4 device in Sonoff rf

What do you mean 16*4? u mean 16 buttons in remotes or switches?

**Comment by LukaszP2** [comment]
> > > in the case of sonoff bridge i think tasmota it's a better fit because you can attach many more rf devices to it vs. the original fw (unless itead changed something meanwhile, i'm not totally up to date) so it's entirely up to you.
> > 
> > 
> > Now it can be 16 x 4 device in Sonoff rf
> 
> What do you mean 16*4? u mean 16 buttons in remotes or switches?

You can add 16 device with 4 swiches. If you add alarm or montion detector is one device.

**Comment by ahmst** [comment]
Thanks for your work, Peter! It would be amazing to have the sonoff rf bridge sending RF codes from HA. Sure you'll achieve it! I use it as a remote for arming and disarming my 433mhz alarm system. If you need someone who test it out your new code, just tell me :)

**Comment by prasanthmk82** [comment]
> Is RF bridge updated to 3.3.0 supported?
> 
> Update: It does support and works well. Thanks @peterbuga
> The alarm sensor is working. Don't know about the button sensor since I don't have one to test.

version 3.3.0 doesn't seem to be working now

**Comment by peterbuga** [comment]
@prasanthmk82 any errors / logs to back that? it'll help explaining **what** exactly is not working anymore

**Comment by ahmst** [comment]
When pressing button of entity from HASSIO i get this error... am I doing anything wrong?
![Captura](https://user-images.githubusercontent.com/36327054/73600434-5c82d600-4550-11ea-82cf-2b2359926563.JPG)

**Comment by peterbuga** [comment]
@ahmst clearly you're doing something wrong.
by definition a **sensor** it's a read-only entity, it just reports data, you cannot turn it on/off :))

**Comment by stefanmechelen** [comment]
any chance for a 3.4 update?

**Comment by tjhiwa** [comment]
Do we have successful progress on RF Bridge for sending signal out? This would be really great and very useful for many of us. I use the RF Bridge remote-buttons to trigger Arm/Disarm of my alarm system.
My sonoff devices aren't sharing the same WiFi SSID (different IP range, too) with HA, so I do not wish to use AlexxIT/SonoffLAN to achieve it.

**Comment by peterbuga** [comment]
> Do we have successful progress on RF Bridge for sending signal out? This would be really great and very useful for many of us. I use the RF Bridge remote-buttons to trigger Arm/Disarm of my alarm system.
> My sonoff devices aren't sharing the same WiFi SSID (different IP range, too) with HA, so I do not wish to use AlexxIT/SonoffLAN to achieve it.

on my last updates, i understood wrong that RF bridge was actually able to copy-send the signals out (stupid i know, but i don't know what made me think that... it happens). as more & more people  steer towards `AlexxIT/SonoffLAN` 's solution (on obvious step and i'm glad i was able to kickstart the integration of these vanilla devices!) i kinda paused the development of this due to obvious reasons. for me it just works better not to stick the devices constantly in lan mode or having them controlled only by HA.

**Comment by tjhiwa** [comment]
> in other words: i can fix it but probably no one (else) will use it

you would be pleasantly surprised by how many actually care and will use it. many don't put their HA on same subnet as their devices (either on different vlan, different router, using repeater, or even hosted elsewhere like in the office or cloud). many don't have multicast available. also, many agree with you on not confining things into only LAN or home HA. 
with what you created here, it allows devices from multiple homes/buildings (network physically separated) to connect together to a single or multiple HA hosted elsewhere (thru ewelink device sharing into one or multiple secondary accounts). different HA users could be separated thru the HA settings to limit which devices are accessible. 
in other words, one to many, many to one, and many to many configuration. AlexxIT would only allow one to one (one account one location). 
the expansion and configuration is wide and forward. it can cater for even corporate or industrial setups. 
the RF bridge integration is the only thing holding people back from going full swing using what you develop here.
please.. please.. fix the RF Bridge integration 🙏

**Comment by SBastrykin** [comment]
This is necessary for me, since the component from AlexxIT / SonoffLAN does not fit me - it hangs a 4g modem in the router
All hope is in you

**Comment by peter-sounthararajah** [comment]
hi all i have a sonoff basic and an rf bridge , i would like HS to be able to receive one of 4 remote signals and trigger different automatons
unfortunately the only one remote entity shows up.
i will leave a copy of the file below.
would someone like to look at it and tell me whats relevant, 
what can be used to create other entities from the RF bridge.
or just give a breakdown on what each part of the code dose what 
kind regards Peter s
[sonoff.docx](https://github.com/peterbuga/HASS-sonoff-ewelink/files/4698457/sonoff.docx)
